src/iso.c

```diff
@@ -0,0 +1,489 @@
+void iso_media_close(struct iso_point *iso_p);
+struct iso_point *iso_media_open(int ata_unit);
+int iso_media_read(void *dest, int elem_size, int num_elem, struct iso_point *stream);
+void iso_media_seek(struct iso_point *iso_p, long offset, int whence);
+
+/**
+ * @brief Reads the next directory record in the extent
+ * @details Starting at the current iso_p, reads a directory record into
+ * the argument dr, moving iso_p forward.
+ *
+ * @param iso_p A pointer to a struct iso_point which is set to the beginning of a directory record in a directory extent
+ * @param dr A pointer to a directory record to be filled with the information
+ * @return 1 on success, 0 on failure
+ */
+int get_directory_record(struct iso_point *iso_p, struct directory_record *dr) {
+    int bytes_read = 0;
+    int max_dr_size = 64;
+    char dr_dest[max_dr_size];
+    int result = iso_media_read(dr_dest, 1, max_dr_size, iso_p);
+    if(result != max_dr_size) {
+        console_printf("get directory record fails\n");
+        return 0;
+    }
+
+    memcpy(&(dr->length_of_record), dr_dest + 0, sizeof(dr->length_of_record));   //1 for bytes read
+    memcpy(&(dr->length_of_ext_record), dr_dest + 1, sizeof(dr->length_of_ext_record));
+    memcpy(dr->loc_of_ext, dr_dest + 2, sizeof(dr->loc_of_ext));
+    memcpy((dr->data_length), dr_dest + 10, sizeof(dr->data_length));
+    memcpy((dr->rec_date_time), dr_dest + 18, sizeof(dr->rec_date_time));
+    memcpy((dr->file_flags), dr_dest + 25, sizeof(dr->file_flags));
+    memcpy((dr->file_flags_interleaved), dr_dest + 26, sizeof(dr->file_flags_interleaved));
+    memcpy(&(dr->interleave_gap_size), dr_dest + 27, sizeof(dr->interleave_gap_size));
+    memcpy((dr->vol_seq_num), dr_dest + 28, sizeof(dr->vol_seq_num));
+    memcpy(&(dr->len_identifier), dr_dest + 32, sizeof(dr->len_identifier));
+    bytes_read += 33;
+
+    memcpy((dr->file_identifier), dr_dest + 33, dr->len_identifier);
+    dr->file_identifier[dr->len_identifier] = '\0';
+    bytes_read += dr->len_identifier;
+
+    iso_media_seek(iso_p, bytes_read - 64, SEEK_CUR);
+    if (bytes_read % 2 == 1) { // Odd length, so we need to seek one bit to pad
+        iso_media_seek(iso_p, 1, SEEK_CUR);
+    }
+    return 1;
+}
+
+/**
+ * @brief Converts a character hex number into a base 10 integer
+ *
+ * @param src Array of hex characters in big endian format
+ * @param len Number of bytes in the src to convert
+ * @return The integer value of the first len bytes of hex characters in src
+ */
+int hex_to_int(char *src, int len) {
+    int i, sum = 0;
+    for (i = 0; i < len; i++) {
+        sum *= 16;
+        sum += src[i];
+    }
+    return sum;
+}
+
+/**
+ * @brief Checks whether flags represent a directory
+ * @details Bit compares directory record file flags to the dir flag bit
+ *
+ * @param flags The file flags from a directory record
+ * @return 1 if dir bit is high, 0 otherwise
+ */
+int is_dir(int flags) {
```

Comment:
I tried replacing it with your one-liner, but the behavior changed and was no longer valid.